### PR TITLE
Add query/reply attachment

### LINKED ISSUE #335: User Attachments
### Describe the feature

User attachments were introduced in Zenoh 0.10.1.

They should now be added to Zenoh Pico.
- [x] Pub/Sub support
- [x] Queryable support
- [x] Reply support

### PR DESCRIPTION
**Pull request by jean-roland**
This PR adds attachments to queries and replies + example. 

It also fixes an attachment-related memory leak.

Closes #335

### CODE REVIEW

File: examples/unix/c11/z_queryable.c [hunk 1]
```diff
@@ -111,6 +146,9 @@ int main(int argc, char **argv) {
 
     printf("Press CTRL-C to quit...\n");
     while (1) {
+        if ((n != 0) && (msg_nb >= n)) {
```
Comment (sashacmc) [inline examples/unix/c11/z_queryable.c:149]:
```suggestion
        if (n != 0 && msg_nb >= n) {
```

File: src/protocol/core.c [hunk 2]
```diff
@@ -125,19 +127,39 @@ int8_t _z_encoded_attachment_iteration_driver(const void *this_, z_attachment_it
...
 int8_t z_attachment_iterate(z_attachment_t this_, z_attachment_iter_body_t body, void *ctx) {
     return this_.iteration_driver(this_.data, body, ctx);
 }
-z_attachment_t z_attachment_null(void) { return (z_attachment_t){.data = NULL, .iteration_driver = NULL}; }
-#endif
\ No newline at end of file
+
+void _z_attachment_copy(z_attachment_t *dst, const z_attachment_t *src) {
+    dst->iteration_driver = src->iteration_driver;
+    // _z_bytes_copy(&dst->data, &src->data);
```
Comment (sashacmc) [inline src/protocol/core.c:150]: Why was this left commented out?

File: tests/modularity.py [hunk 3]
```diff
@@ -4,10 +4,20 @@
...
 import sys
 import time
+import difflib
 
 # Specify the directory for the binaries
 DIR_EXAMPLES = "build/examples"
 
+def print_string_diff(str_a, str_b):
+    for i,s in enumerate(difflib.ndiff(str_a, str_b)):
+        if s[0]==' ': continue
```
Comment (sashacmc) [inline tests/modularity.py:14]: Nit: please use some auto-formatting tool for the python, next time if this file will saved by me or others with auto-format configured, there will be unnecessary diff

**Review by sashacmc — CHANGES_REQUESTED** [review]

File: examples/unix/c11/z_get.c [hunk 4]
```diff
@@ -116,6 +130,11 @@ int main(int argc, char **argv) {
     if (value != NULL) {
         opts.value.payload = _z_bytes_wrap((const uint8_t *)value, strlen(value));
     }
+#if Z_FEATURE_ATTACHMENT == 1
+    z_owned_bytes_map_t map = z_bytes_map_new();
+    z_bytes_map_insert_by_alias(&map, _z_bytes_wrap((uint8_t *)"hi", 2), _z_bytes_wrap((uint8_t *)"there", 5));
```
[the remark this answers is not in the document]
Comment (jean-roland) [reply examples/unix/c11/z_get.c:135]: Valid point, I just pasted what we had on the pub/sub attachment example, I didn't pay attention there was a `_z_*` function in it.

File: examples/unix/c11/z_queryable.c [hunk 5]
```diff
@@ -21,29 +21,60 @@
...
+    printf(">> [Queryable handler] Received Query '%s?%.*s'\n", z_loan(keystr), (int)pred.len, pred.start);
     if (payload_value.payload.len > 0) {
         printf("     with value '%.*s'\n", (int)payload_value.payload.len, payload_value.payload.start);
     }
+#if Z_FEATURE_ATTACHMENT == 1
+    if (z_attachment_check(&query->_val._rc.in->val.attachment)) {
+        printf("Attachement found\n");
+        z_attachment_iterate(query->_val._rc.in->val.attachment, attachment_handler, NULL);
+    }
+    z_attachment_drop(&((z_query_t *)query)->_val._rc.in->val.attachment);
```
[the remark this answers is not in the document]
Comment (jean-roland) [reply examples/unix/c11/z_queryable.c:48]: Hmm yeah, you're right, it should be dropped at the same time as the query.

File: src/protocol/core.c (hunk 2 above)
Comment (jean-roland) [reply src/protocol/core.c:150]: Hmm this looks like a mistake.

File: examples/unix/c11/z_queryable.c (hunk 1 above)
Comment (jean-roland) [reply examples/unix/c11/z_queryable.c:149]: Is there a particular reason? I prefer when there is no ambiguity for operation order.

Comment (sashacmc) [reply examples/unix/c11/z_queryable.c:149]: looks very redundant, and the order of operations is strictly defined, but if you like it better, you can leave it.

**Review by sashacmc — APPROVED** [review]
LGTM

**Review by Mallets — APPROVED** [review]
